autoload/neomake/makers/ft/javascript.vim

```diff
@@ -1,11 +1,29 @@
 " vim: ts=4 sw=4 et
 
 function! neomake#makers#ft#javascript#EnabledMakers() abort
-    return ['jshint', 'jscs',
+    return ['jshint', 'jscs', 'tsc',
                 \ executable('eslint_d') ? 'eslint_d' : 'eslint',
                 \]
 endfunction
 
+function! neomake#makers#ft#javascript#tsc() abort
+    " tsc should not be passed a single file.
+    let maker = {
+        \ 'args': ['--noEmit', '--watch', 'false', '--pretty', 'false'],
+        \ 'append_file': 0,
+        \ 'errorformat':
+            \ '%E%f %#(%l\,%c): error %m,' .
+            \ '%E%f %#(%l\,%c): %m,' .
+            \ '%Eerror %m,' .
+            \ '%C%\s%\+%m'
+        \ }
+    let config = neomake#utils#FindGlobFile('tsconfig.json')
+    if !empty(config)
+        let maker.args += ['--project', config]
+    endif
+    return maker
+endfunction
```

Comment:
Just use `return neomake#makers#ft#typescript#tsc()` here?

Reply:
ok. I've changed the code now is ready :+1: